### PR TITLE
formatRequestError

### PR DESCRIPTION
**Pull request by u9520107**
Modified RelayDefaultNetworkLayer's formatRequestErrors function to check for locations array before attempting to call locations.map.

#325

### CODE REVIEW

File: src/network-layer/default/RelayDefaultNetworkLayer.js [hunk 1]
```diff
@@ -183,16 +183,21 @@ function formatRequestErrors(
...
           queryLine.substr(column - 1 - offset, CONTEXT_LENGTH),
           ' '.repeat(offset) + '^^^'
         ].map(messageLine => indent + messageLine).join('\n');
-      }).join('\n')
+        }).join('\n'))
+      : '';
+
+    return (
+      prefix + message + locationMessage
     );
```
Comment (josephsavona) [inline src/network-layer/default/RelayDefaultNetworkLayer.js]: `return prefix + message + locationMessage;`

File: src/network-layer/default/RelayDefaultNetworkLayer.js [hunk 2]
```diff
@@ -183,16 +183,21 @@ function formatRequestErrors(
   return errors.map(({locations, message}, ii) => {
     var prefix = (ii + 1) + '. ';
     var indent = ' '.repeat(prefix.length);
-    return (
-      prefix + message + '\n' +
-      locations.map(({column, line}) => {
+
+    //custom errors thrown in graphql-server may not have locations
+    var locationMessage = locations
```
Comment (josephsavona) [inline src/network-layer/default/RelayDefaultNetworkLayer.js]: style nit, but we prefer the following style for ternaries:

```javascript
var locationMessage = locations ?
  foo :
  bar;
```

**Comment by josephsavona** [comment]
@u9520107 thanks for the contribution!

**Comment by u9520107** [comment]
Fixed coding styles accordingly.

**Comment by u9520107** [comment]
err.. not sure why the travis-ci build failed.. 
Anything else I should change?

**Comment by josephsavona** [comment]
@u9520107 it looks like the failure was related to a flow update and not related to your PR. Thanks for contributing!

**Pull request closed without merging**